### PR TITLE
[4.1 ScheduledTasks] changes

### PR DESCRIPTION
**Pull request by niharikamahajan02**
Pull Request for Issue #36453  .

### CODE REVIEW

**Comment by brianteeman** [comment]
image

File: administrator/language/en-GB/com_scheduler.ini [hunk 1]
```diff
@@ -121,17 +129,11 @@ COM_SCHEDULER_TASK_PRIORITY_ASC="Task Priority ascending"
 COM_SCHEDULER_TASK_PRIORITY_DESC="Task Priority descending"
 COM_SCHEDULER_TASK_ROUTINE_EXCEPTION="Routine threw exception: %1$s"
 COM_SCHEDULER_TASK_TYPE="Task Type"
+COM_SCHEDULER_TASK_TYPE_ASC="Task Type ascending"
```
Comment (tecpromotion) [inline administrator/language/en-GB/com_scheduler.ini:132]:
```suggestion
COM_SCHEDULER_TASK_TYPE_ASC="Task type ascending"
```

File: administrator/language/en-GB/com_scheduler.ini [hunk 2]
```diff
@@ -121,17 +129,11 @@ COM_SCHEDULER_TASK_PRIORITY_ASC="Task Priority ascending"
...
+COM_SCHEDULER_TASK_TYPE_DESC="Task Type descending"
```
Comment (tecpromotion) [inline administrator/language/en-GB/com_scheduler.ini:133]:
```suggestion
COM_SCHEDULER_TASK_TYPE_DESC="Task type descending"
```

File: administrator/language/en-GB/com_scheduler.ini [hunk 3]
```diff
@@ -121,17 +129,11 @@ COM_SCHEDULER_TASK_PRIORITY_ASC="Task Priority ascending"
 COM_SCHEDULER_TASK_PRIORITY_DESC="Task Priority descending"
 COM_SCHEDULER_TASK_ROUTINE_EXCEPTION="Routine threw exception: %1$s"
 COM_SCHEDULER_TASK_TYPE="Task Type"
```
Comment (tecpromotion) [inline administrator/language/en-GB/com_scheduler.ini:131]:
```suggestion
COM_SCHEDULER_TASK_TYPE="Task type"
```

File: administrator/language/en-GB/com_scheduler.ini (hunk 1 above)
Comment (tecpromotion) [inline administrator/language/en-GB/com_scheduler.ini:132]:
```suggestion
```

File: administrator/language/en-GB/com_scheduler.ini (hunk 2 above)
Comment (tecpromotion) [inline administrator/language/en-GB/com_scheduler.ini:133]:
```suggestion
```

File: administrator/language/en-GB/com_scheduler.ini [hunk 4]
```diff
@@ -121,17 +129,11 @@ COM_SCHEDULER_TASK_PRIORITY_ASC="Task Priority ascending"
...
+COM_SCHEDULER_TASK_TYPE_ASC="Task Type ascending"
+COM_SCHEDULER_TASK_TYPE_DESC="Task Type descending"
+
```
Comment (tecpromotion) [inline administrator/language/en-GB/com_scheduler.ini]:
```suggestion
```

File: administrator/components/com_scheduler/tmpl/tasks/default.php [hunk 5]
```diff
@@ -226,7 +226,7 @@ class="js-draggable" data-url="<?php echo $saveOrderingUrl; ?>" data-direction="
 						<td class="small d-none d-md-table-cell">
 							<button type="button" class="btn btn-sm btn-warning" <?php echo $item->state < 0 ? 'disabled' : ''; ?> data-id="<?php echo (int) $item->id; ?>" data-title="<?php echo htmlspecialchars($item->title); ?>" data-bs-toggle="modal" data-bs-backdrop="static" data-bs-target="#scheduler-test-modal">
```
Comment (tecpromotion) [inline administrator/components/com_scheduler/tmpl/tasks/default.php]:
```suggestion
							<button type="button" class="btn btn-sm btn-warning" <?php echo $item->state < 0 ? 'disabled' : ''; ?> data-id="<?php echo (int) $item->id; ?>" data-title="<?php echo htmlspecialchars($item->title); ?>" data-bs-toggle="modal" data-bs-backdrop="static" data-bs-target="#scheduler-task-modal">
```

Comment (tecpromotion) [reply administrator/components/com_scheduler/tmpl/tasks/default.php]: Please check if this works

File: administrator/language/en-GB/com_scheduler.ini (hunk 3 above)
Comment (brianteeman) [reply administrator/language/en-GB/com_scheduler.ini:131]: Task Type is correct in the same way as we have Task Priority

File: composer.json [hunk 6]
```diff
@@ -18,6 +18,7 @@
         "allow-plugins": {
             "dealerdirect/phpcodesniffer-composer-installer": true
         }
+
```
Comment (tecpromotion) [inline composer.json]:
```suggestion
```

File: administrator/components/com_scheduler/access.xml [hunk 7]
```diff
@@ -8,12 +8,12 @@
 		<action name="core.delete" title="JACTION_DELETE" />
 		<action name="core.edit" title="JACTION_EDIT" />
 		<action name="core.edit.state" title="JACTION_EDITSTATE" />
-		<action name="core.testrun" title="COM_SCHEDULER_PERMISSION_TESTRUN" />
+		<action name="core.runtask" title="COM_SCHEDULER_PERMISSION_RUN_TASK" />
```
Comment (tecpromotion) [inline administrator/components/com_scheduler/access.xml:11]: Please check https://github.com/joomla/joomla-cms/blob/53544c6a58778cfe9e338d0ba80149c67bb5452d/plugins/system/schedulerunner/schedulerunner.php#L267

File: administrator/components/com_scheduler/tmpl/tasks/default.php (hunk 5 above)
Comment (tecpromotion) [reply administrator/components/com_scheduler/tmpl/tasks/default.php]: Please check

https://github.com/joomla/joomla-cms/blob/53544c6a58778cfe9e338d0ba80149c67bb5452d/build/media_source/com_scheduler/js/admin-view-run-test-task.es6.js#L22
and
https://github.com/joomla/joomla-cms/blob/53544c6a58778cfe9e338d0ba80149c67bb5452d/administrator/components/com_scheduler/tmpl/tasks/default.php#L253

**Review by tecpromotion — CHANGES_REQUESTED** [review]
Please check my comments @niharikamahajan02

File: build/media_source/com_scheduler/js/admin-view-run-test-task.es6.js [hunk 8]
```diff
@@ -42,7 +42,9 @@ const initRunner = () => {
     const id = parseInt(button.dataset.id, 10);
     const { title } = button.dataset;
 
-    modal.querySelector('.modal-title').innerHTML = Joomla.Text._('COM_SCHEDULER_TEST_RUN_TITLE').replace('%d', id.toString());
+
+    modal.querySelector('.modal-title').innerHTML = Joomla.Text._('COM_SCHEDULER_RUN_TASK_TITLE').replace('%d', id.toString());
+
```
Comment (PhilETaylor) [inline build/media_source/com_scheduler/js/admin-view-run-test-task.es6.js]: This should fix the drone complaining 

```suggestion
    modal.querySelector('.modal-title').innerHTML = Joomla.Text._('COM_SCHEDULER_RUN_TASK_TITLE').replace('%d', id.toString());

```

**Comment by richard67** [comment]
@niharikamahajan02 The branch for this PR meanwhile has conflicts due to recent changes in the 4.1-dev branch. You can try to solve the conflicts, but maybe it's easier to make a new PR. In both cases you should update your 4.1-dev branch to the same branch of the upstream repository so it is up to date. Let us know if you need help or advise.

**Comment by niharikamahajan02** [comment]
Ok, I will make a new PR then. Thanks

**Comment by tecpromotion** [comment]
And please look for any occurrences of old strings. I have found a few more. 

e.g.

media/com_scheduler/js/admin-view-run-test-task.js
media/com_scheduler/js/admin-view-run-test-task-es5.js

File: administrator/language/en-GB/com_scheduler.ini [hunk 9]
```diff
@@ -123,14 +131,6 @@ COM_SCHEDULER_TASK_ROUTINE_EXCEPTION="Routine threw exception: %1$s"
...
 COM_SCHEDULER_TASK_TYPE_DESC="Task Type descending"
-COM_SCHEDULER_TEST_RUN="Run Test"
-COM_SCHEDULER_TEST_RUN_DURATION="Duration: %s seconds"
-COM_SCHEDULER_TEST_RUN_OUTPUT="Output:<br>%s"
-COM_SCHEDULER_TEST_RUN_STATUS_COMPLETED="Status: Completed"
-COM_SCHEDULER_TEST_RUN_STATUS_STARTED="Status: Started"
-COM_SCHEDULER_TEST_RUN_STATUS_TERMINATED="Status: Terminated"
-COM_SCHEDULER_TEST_RUN_TASK="Task: \"%s\""
-COM_SCHEDULER_TEST_RUN_TITLE="Test task (ID: %d)"
 COM_SCHEDULER_TEST_TASK="Test Task"
```
Comment (tecpromotion) [inline administrator/language/en-GB/com_scheduler.ini:134]:
```suggestion
```

**Comment by niharikamahajan02** [comment]
> And please look for any occurrences of old strings. I have found a few more.
> 
> e.g.
> 
> media/com_scheduler/js/admin-view-run-test-task.js media/com_scheduler/js/admin-view-run-test-task-es5.js

Ok sir.

**Comment by niharikamahajan02** [comment]
> And please look for any occurrences of old strings. I have found a few more.
> 
> e.g.
> 
> media/com_scheduler/js/admin-view-run-test-task.js media/com_scheduler/js/admin-view-run-test-task-es5.js

I tried changing all the constants in -

media/com_scheduler/js/admin-view-run-test-task.js
media/com_scheduler/js/admin-view-run-test-task-es5.js

and also in other files , but one problem is that even after changing constants in minified js also (in my code editor)  that change is not getting reflected in 
[media/com_scheduler/js/admin-view-run-test-task.min.js](http://localhost/joomla/joomla-cms/media/com_scheduler/js/admin-view-run-test-task-es5.min.js?6281cc2e15f16e34efa477c4e29c4976)   like
image

**Comment by brianteeman** [comment]
Do not touch anything in the /media folder

The changes must be made in /build/media-source

Then you run the npm task to rebuild the /media folder (`npm run build:js`)

**Comment by richard67** [comment]
See also https://docs.joomla.org/J4.x:Setting_Up_Your_Local_Environment

**Comment by niharikamahajan02** [comment]
Okay, thanks!

**Comment by niharikamahajan02** [comment]
Yes , the problem is now fixed I'll just submit a new PR now.

**Pull request closed without merging**